### DISCUSSION TITLE
G7 and mismatch

### DISCUSSION THREAD

**Discussion by kakoni** [NightscoutFoundation/xDrip#3294] [Q&A] [answered]
Hi. I've run several succesful G7 sessions with xdrip but now I could not get the new sensor to pair.
I'm getting MISMATCH error. 

Things that I've done;
1) Well obviously verified that the sensor/transmitter code and xdrip transmitter id are the same
2) Restarted collector
3) Change transmitter id to somethign else and then back to the real one
4) Restarted phone

No luck.

**Comment by Navid200** [comment, marked as answer]
I have experienced this same issue. And I can recreate it as well.  
I have test phones that I use to monitor the voltages of my used G7 devices because I am creating a log of how fast the battery voltages drop.  It happens all the time when I keep switching the transmitter ID.  
I tell you what I do and I will also tell the the side effect.  Please read everything.

A guaranteed solution is to uninstall xDrip (a bad thing) and reinstall.  
Obviously, after you reinstall, you will need to scan the QR code again.
I say uninstalling xDrip is bad because if you uninstall xDrip, you will lose the database and all your settings.
You can create a backup of both.  
But, my database is not very important to me because I always upload my readings to Nightscout.  So, if I lose my xDrip database, it does not matter.
But, I have settings where I have my alerts and display settings etc.
So, I backup my settings before uninstall.

This is something I hope we will improve.  But, I cannot guarantee we can.

@jamorham

**Comment by kakoni** [comment]
So ok, its something in the keks library I believe,
https://github.com/NightscoutFoundation/xDrip/blob/1af5128f5b083cb6406661d5b9987fe7173698dd/libkeks/src/main/java/jamorham/keks/Plugin.java#L239

**Comment by jamorham** [comment]
Move your old G7 out of range should fix it or it is supposed to resolve by itself if you just wait.

**Reply by Navid200** [reply]
My comment about being able to recreate the problem does not take this into account.

When I use my test phone to pair with my old G7 devices, my current device is on my arm and I cannot place it inside the microwave.  Therefore, my statement does not contradict what JamOrHam is suggesting.

**Reply by kakoni** [reply]
@jamorham Old G7 was away for over an hour. I didnt get the pairing request during that time, just this mismatch error.
Would it be advisable to clean all the existing bluetooth dexcom connections from android bluetooth setting before starting a with new sensor?

Also another question; Does G7 actually stop sensor when you click stop sensor on xdrip? Or is it so that it keeps transmitting?

**Comment by jamorham** [comment]
@kakoni which xDrip version are you using?

**Reply by kakoni** [reply]
@jamorham I'm in using the nightly from 2023.12.16.

**Reply by Navid200** [reply]
@kakoni Would you please post a screenshot of the classic status page?  
https://navid200.github.io/xDrip/docs/xDrip-Version.html

**Comment by jamorham** [comment]
@kakoni that xdrip version should be fine. Double check that your transmitter number is a number, if there is 0 then it is a 0 not an O and ensure that any previously used sensors are a long way away as they can be the source of mismatch errors.

**Reply by Navid200** [reply]
There was no xDrip released on December 16.

**Reply by kakoni** [reply]
@Navid200 I'm on the nightly version, https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.12.16

**Reply by Navid200** [reply]
I see.  My mistake.  Sorry about that.

**Comment by Gonzmdv** [comment]
Everything was going fine with my previous sensor and I'm having the mismatch problem with the new G7.
I have uninstalled, installed and scanned the code, adjusted the settings following the instructions, rebooted the phone, introduced the code, the phone paired immediately but I don't get any reading. The Bluetooth devices also shows the new sensor. The pump is reading the G7 with no issues. I can't see any problem in the status page:
![Screenshot_20240123-004812](https://github.com/NightscoutFoundation/xDrip/assets/45602955/1fcd4643-b965-4085-9a45-92cd11216fca)
Any idea of what could be happening?
Thanks in advance.

**Reply by Navid200** [reply]
If you check under the main menu, do you see a stop option or a start option?

**Reply by nsghub** [reply]
I just ran into the same problem.
The solution was to change the ID again. I think the Start option was available, but not 100% sure. The missing data is not backfilled.

**Comment by Gonzmdv** [comment]
Sorry, I went to bed and, at some point, it started reading. I'm sure there was not a stop option, but I'm not sure if there was a start option.
Before re-installing xDrip there was the mismatch word and stop sensor, so I tried to change to a false paring code, stop sensor and enter the true pairing code again, but it didn't work, so I finally reinstalled when I read this thread. I hope you can fix it in a future. What is happening?

**Reply by Navid200** [reply]
You should never stop a G7.  There is no reason to do that.
I wanted to remove the stop option from xDrip.  But, I found out that it also existed in the Dexcom app.  So, I did not proceed.  In my opinion, it is a design mistake by Dexcom to include it.

**Reply by kakoni** [reply]
@Navid200 Interesting, I've (out-of-the-old g6 habit) pressed the stop button in xdrip. Lets ask the question otherway; Is there some harm by doing 
- stop (g7) sensor
- Insert new
- start sensor?

**Reply by Navid200** [reply]
No, there is no harm in doing that.

I didn't want the option in xDrip because there are individuals who are used to Libre and they just stop sensor as a troubleshooting step.  They just stop and then start.
And I fear one of these days, someone stops G7 and then finds out they should not have done that when it's too late.

I can not imagine a scenario where one may need to stop a G7.

I am not sure exactly how Dexcom comes up with the Bluetooth name for a G7.
With G6, it was Dexcom followed by the last two characters of the G6 transmitter serial number.
But, with G7, the two characters in the Bluetooth name specific to each device have no correlation to the pairing code.
So, I don't know under what circumstances the Bluetooth names of your two consecutive devices would be the same.
If you face that scenario and you want to switch to your new device, you will not be able to connect to it as long as your old device is still present and not placed inside the microwave.  Stopping it is not going to change that.  A G7 that has expired continues to provide a Bluetooth signal.  I am creating a graph of the battery voltages for my G7 devices.  The one I started using almost 3 months ago is still providing battery voltages over Bluetooth.

You gain nothing by stopping a G7.

**Reply by Navid200** [reply]
And I'm not sure why you do that unless you use each device for 10.5 days to extend the run time by 5%.

I insert my new device always in the morning.  But, I leave the old one running for 12 remaining hours and I continue to use the old one in xDrip.
I just use a test phone to connect to the new one after it warms up to confirm that it is working.
But, after that, I am not connected to the new device for 12 hours.
When the old one ends, I just switch the id in my main xDrip and it switches right away.
This lets me start each session with a device that has been running for 12 hours already hence provide more stable readings.

I have gone over this here: https://navid200.github.io/xDrip/docs/Dexcom/G7_Grace.html

**Comment by Gonzmdv** [comment]
I thought that maybe xDrip was not able to detect the new sensor because the old G7 session was still running.
What happens if the new sensor has the same code?

**Reply by jamorham** [reply]
Put removed sensors a long way away (out of bluetooth range) is probably the best advice. I wouldn't issue stop sensor as there is a slim chance you end up stopping your new one.

**Comment by kakoni** [comment]
So just started new session (3 days, early, old one died in sensor error). Based on discussion here's  what I did
1) Removed the old sensor
2) Placed the old sensor (and any old sensor for that matter) far away (Or perhaps microwave owen would do fine aswell?)
3) Did NOT stop the sensor (from xdrip)
4) Inserted new sensor 
5) Changed the tranmitter ID in xdrip
6) Went to G7 status page to hunt for the transmitter
7) Approved pairing request

No problems.

**Reply by Gonzmdv** [reply]
kakoni, I would like to know if your experience with the G7s is being better o worse than with the G6s.
The hospital gave me three G7s to try. The first ones were a complete disaster. The initial value was far away from my real glucose level and they were frequently loosing the Bluetooth connection. The second sensor didn't last for the whole 10 days. However, the last G7 was accurate from the first reading and did not lose any reading.
Now, I have been asked if I want to keep my G6s or change to the G7s and I don't know what to do. Not having warming time is great, but if most G7s miss readings so frequently they are not an option for me.
Thank you.

**Reply by kakoni** [reply]
@Gonzmdv I havent experienced bluetooth connection problems. I'm using AAPS with exponential smoothing for the BG values.

**Reply by Gonzmdv** [reply]
What is AAPS with exponential smoothing for the BG values?

**Reply by kakoni** [reply]
@Gonzmdv Aah sorry.

1.  AAPS=AndroidAPS (https://androidaps.readthedocs.io/en/latest/index.html) 
2. Exponential smoothing=BG value smoothing, because "Noteworthy is the fact that the G7 system, compared to the G6, does not smooth the values, neither in the app, nor in the reader." See https://androidaps.readthedocs.io/en/latest/Hardware/DexcomG7.html and https://androidaps.readthedocs.io/en/latest/Usage/Smoothing-Blood-Glucose-Data.html#exponential-smoothing

But this these are related to androidaps. Now in the xdrip world, I'm not sure about the g7 smoothing, @Navid200 is xdrip displaying just the raw data from the sensor or is there some smoothing algorithm that can be turned on?

**Reply by Navid200** [reply]
The phrase "raw data" may not be the best phrase to use because it could be confused with what a G5 transmitter did, which was to transmit what the G5 reiver showed as well as data before calibration had been applied to it.  That is what we called raw data.  

Neither G6 nor G7 transmits raw data.
xDrip shows what is transmitted by the G6 or G7 transmitter.  

xDrip offers graph smoothing.   https://navid200.github.io/xDrip/docs/Display/GraphSmoothing.html

**Comment by Gonzmdv** [comment]
There is a Smoothing option under Display Settings Menu in xDrip+, but I'm not sure how it works.
I've been wearing both G6 ang G7 at the same time, and G7s goes up and down faster than G6. The predictive side of the algorithm has more weight in G7s than in G6s.

**Reply by kakoni** [reply]
@Gonzmdv Have a look at this article from september. Interesting read; https://danheller.substack.com/p/the-dexcom-g7-vs-g6-which-is-better

**Reply by Gonzmdv** [reply]
Thank you very much! My experience has been the same. I will keep the G6.

**Comment by rpm5099** [comment]
I'm stuck on Mismatch - wait/scanning errors.  Last sensor I ended up reinstalling about 5 times but reloading the settings would cause it to not connect, so I recreated all of the settings and it finally worked.  This time, even after doing that I'm still stuck with the same status.  The old sensor is nowhere near and it's in an electrostatic bag.  This is a painful process to go through for every new sensor, and right now I can't get this one to connect at all.  I'll try with a test phone tomorrow.  I'm on nightly build from Nov 8th 2024, have configured all of the recommended settings and scanned the qr code.  Samsung galaxy s23 ultra.  I appreciate any ideas.

**Reply by Navid200** [reply]
I would start a new thread and include a screenshot of the status page.
There is no reason to bring out this old thread.  A lot has changed since then.  I never need to reinstall xDrip any longer.

**Reply by rpm5099** [reply]
Ok, I will do that. This looked like the same problem.

**Comment by nsghub** [comment]
I was stuck on Sunday too. I ended up deleting xdrip and installing it again. I am now on the version from 3rd November.
Don't forget to install the QR code again https://navid200.github.io/xDrip/docs/Dexcom/G7.html

18.11.2024 16:06:23 Duke McCoy ***@***.***>:
